### PR TITLE
JAMES-2865 Enhance builders for MockSMTPBehavior

### PR DESCRIPTION
**Pull request by chibenwa**

### CODE REVIEW

File: server/mailet/integration-testing/src/test/java/org/apache/james/mailets/RemoteDeliveryErrorTest.java [hunk 1]
```diff
@@ -250,7 +179,13 @@ public void remoteDeliveryShouldBounceWhenAlwaysRCPT421() throws Exception {
     @Test
     public void remoteDeliveryShouldBounceWhenAlwaysFROM421() throws Exception {
         with()
-            .body(ALWAYS_421_FROM_BEHAVIOR)
+            .body(MockSmtpBehaviors.builder()
```
Comment (mbaechler) [inline server/mailet/integration-testing/src/test/java/org/apache/james/mailets/RemoteDeliveryErrorTest.java]: I will keep being annoying but the use of rest-assured makes things more painful than they could be.
By delivering a full API, we could get rid of the explicit builder-to-body conversion. We could also avoid repeating the endpoint.
Finally, we could check the server return status without repeating ourselves.

Comment (chibenwa) [reply server/mailet/integration-testing/src/test/java/org/apache/james/mailets/RemoteDeliveryErrorTest.java]: @trantienduchn let's go for your Feign API. Can you take care of it? (next week?)

Comment (trantienduchn) [reply server/mailet/integration-testing/src/test/java/org/apache/james/mailets/RemoteDeliveryErrorTest.java]: Okay, That's fine for me!

**Comment by chibenwa** [comment]
test this please

File: server/mailet/integration-testing/src/test/java/org/apache/james/mailets/RemoteDeliveryErrorTest.java (hunk 1 above)
Comment (trantienduchn) [reply server/mailet/integration-testing/src/test/java/org/apache/james/mailets/RemoteDeliveryErrorTest.java]: #2675

**Comment by chibenwa** [comment]
Merged.

**Pull request closed without merging**